### PR TITLE
Rename authz to authorization for greater clarity

### LINKED ISSUE #19213: Proposal: Rename `--authz-plugin` to `--authorization-plugin`
I must admit that I stumbled upon the term `AuthZ Plugin` in the documentation before, but was confused about what it meant. Based on my personal experience, I would therefore suggest that other users have similar problems.

I wanted to note that using an unsually cased acronym like `AuthZ` seems unsual for Docker.
This goes especially for the CLI, which normally seems to prefer fully spelled-out words.

The documentation under [`docs/extend/authorization.md`](https://github.com/docker/docker/blob/master/docs/extend/authorization.md) always uses the term `authorization plugin` when referring to such plugins in prose. It is therefore not clear to me why the Plugin API and the CLI should use the term `AuthZ Plugin`.

(Disclaimer: I mainly dislike the term `AuthZ`, which seems to suggest some specific technology, and would suggest using either `Auth` or `Authorization` instead of it. This would also remove some issues around casing, like in the URL for the endpoint `/AuthzPlugin.AuthZReq`).

----

**References:** PR #15365 introduced the current authorization plugin infrastructure. The general issue for user access control in Docker is #14674.

### PR DESCRIPTION
**Pull request by tiborvass**
Signed-off-by: Tibor Vass tibor@docker.com

Fixes #19213

### CODE REVIEW

File: api/server/server.go [hunk 1]
```diff
@@ -30,14 +30,14 @@ const versionMatcher = "/v{version:[0-9.]+}"
...
-	CorsHeaders      string
-	AuthZPluginNames []string
-	Version          string
-	SocketGroup      string
-	TLSConfig        *tls.Config
-	Addrs            []Addr
+	Logging                  bool
+	EnableCors               bool
+	CorsHeaders              string
+	AuthorizationPluginNames []string
```
Comment (tiborvass) [inline api/server/server.go]: on a sidenote, @calavera don't you think we should name this `AuthorizationPlugins` instead, if we're making this part of the config file eventually? I don't like the `Names` part.

**Comment by runcom** [comment]
LGTM assuming grep isn't wrong :)

**Comment by tiborvass** [comment]
@runcom I did skip pkg/authorization

**Comment by runcom** [comment]
@tiborvass I thought that's fine

**Comment by jfrazelle** [comment]
```
api/server/middleware.go:172: s.cfg.AuthZPluginNames undefined (type *Config has no field or method AuthZPluginNames)
00:41:36 api/server/middleware.go:173: s.cfg.AuthZPluginNames undefined (type *Config has no field or method AuthZPluginNames)
```

**Comment by tiborvass** [comment]
@jfrazelle yeah i was too excited but i fixed it right before your comment ;)

**Comment by jfrazelle** [comment]
sweet :)

File: api/server/server.go (hunk 1 above)
Comment (calavera) [inline api/server/server.go]: the configuration file uses the flag name, not the struct internal name, so it'd be `authorization-plugin`:

https://github.com/docker/docker/pull/18587#issuecomment-170665246

Comment (tiborvass) [inline api/server/server.go]: oh that's right, that's cool :)

**Comment by thaJeztah** [comment]
Might wanna use GitHub as well for searching; https://github.com/docker/docker/search?utf8=✓&q=authz, e.g.
- https://github.com/docker/docker/blob/38440cadf3e27b32b3ee7b9011b1a0a87cbcca69/integration-cli/docker_cli_authz_unix_test.go#L21-L22
- https://github.com/docker/docker/blob/b714e03fddc71b0d1af7a27198ae5a342b8ce39f/pkg/authorization/authz.go#L14-L25

**Comment by tiborvass** [comment]
@thaJeztah that's on purpose: https://github.com/docker/docker/pull/19279#issuecomment-171112270

**Comment by thaJeztah** [comment]
oh, so sorry, overlooked that comment; don't you think it may be confusing? (not a blocker, just asking)

**Comment by tiborvass** [comment]
@thaJeztah i dont think so, because they are all implementation details users can't see. Except for 2 debug message in pkg/authorization. I decided it's fine and instead added a note explaining what AuthZ and AuthN mean in the docs.

**Comment by thaJeztah** [comment]
sgtm, thx

**Comment by tonistiigi** [comment]
@tiborvass you forgot `integration-cli`

**Comment by vdemeester** [comment]
@tiborvass not in docs-review yet, but you missed something in `man/docker-daemon.8.md` : see https://github.com/tiborvass/docker/blob/rename-authz-to-authorization/man/docker-daemon.8.md#access-authorization. Same in ` docs/reference/commandline/daemon.md` (`Access Authorization`)

**Comment by tiborvass** [comment]
@vdemeester nice catch will update

**Comment by LK4D4** [comment]
LGTM

**Comment by tiborvass** [comment]
Ping @vdemeester @thaJeztah @MHBauer

File: docs/extend/authorization.md [hunk 2]
```diff
@@ -45,6 +45,9 @@ Authorization plugins must follow the rules described in [Docker Plugin API](plu
 Each plugin must reside within directories described under the 
 [Plugin discovery](plugin_api.md#plugin-discovery) section.
 
+Note: the abbreviations `AuthZ` and `AuthN` mean authorization and authentication
```
Comment (thaJeztah) [inline docs/extend/authorization.md]: Should be formatted as note;

```Markdown
> **Note**: the abbreviations ....
```

Comment (thaJeztah) [inline docs/extend/authorization.md]: Also, do we mention `AuthN` already in our docs? (just a nit)

**Comment by thaJeztah** [comment]
one nit for formatting, LGTM otherwise

File: docs/extend/authorization.md (hunk 2 above)
Comment (tiborvass) [inline docs/extend/authorization.md]: Yes we do mention `AuthN` in the API `UserAuthNMethod` in `docs/extend/authorization.md`

**Comment by tiborvass** [comment]
Fixed

**Comment by tiborvass** [comment]
this was green already, merging

**Pull request merged**